File: samples/java/src/main/java/co/windly/limbosample/presentation/main/root/RootComponent.java
package co.windly.limbosample.presentation.main.root;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface RootComponent extends AndroidInjector<RootFragment> {

  //region Factory

  @Subcomponent.Factory
  interface Factory extends AndroidInjector.Factory<RootFragment> {
    // No-op.
  }

  //endregion
}

File: examples/mvp-kotlin/src/main/java/co/windly/limbosample/presentation/main/root/RootComponent.kt
package co.windly.limbosample.presentation.main.root

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface RootComponent : AndroidInjector<RootFragment> {

  //region Factory

  @Subcomponent.Factory
  interface Factory : AndroidInjector.Factory<RootFragment>

  //endregion
}
